src/dcp/dcp.c

```diff
@@ -451,36 +448,25 @@ int main(int argc, char** argv)
             rc = 1;
         }
 
-        /* free the file list */
-        mfu_flist_free(&flist);
-
         /* free the path parameters */
         mfu_param_path_free_all(numpaths, paths);
 
         /* free memory allocated to hold params */
         mfu_free(&paths);
-
-        /* free the input file name */
-        mfu_free(&inputname);
     } 
 #ifdef DAOS_SUPPORT
     /* Perform an object-level copy for DAOS types */
-    /* TODO consider moving most of this into mfu_daos as a single call.
-     * The benefit would be that we could reuse it for dsync. */
     else {
         /* take a snapshot and walk container to get list of objects,
          * returns epoch number of snapshot */
-        int tmp_rc = mfu_flist_walk_daos(daos_args, flist);
+        int tmp_rc = mfu_daos_flist_walk(daos_args, daos_args->src_coh,
+                                         &daos_args->src_epc, flist);
         if (tmp_rc != 0) {
             rc = 1;
         }
 
-        /* all objects are on rank 0 at this point,
-         * evenly spread them among the ranks */
-        mfu_flist newflist = mfu_flist_spread(flist);
-
-        /* perform copy after oids are spread evenly across all ranks */
-        tmp_rc = mfu_flist_copy_daos(daos_args, newflist);
+        /* Collectively copy all objects */
+        tmp_rc = mfu_daos_flist_sync(daos_args, flist, false, true);
```

Comment:
@daltonbohning maybe I missed it, but in this call I am not seeing where the destination gets updated for the copy after the recx are listed using this function?

Reply:
@dsikich I'm not sure I understand the question. `mfu_daos_flist_sync` will iterate over the `recxs` and copy to the destination.

Reply:
@daltonbohning yeah, I guess I was looking for the call to obj_update on the destination handle but I maybe missed seeing it.

Reply:
@dsikich Ah, yes. I think maybe I should abstract the `mfu_daos_flist_sync_recx_array` function into multiple smaller functions so it's not so cluttered.